.github/workflows/snap.yaml

```diff
@@ -16,195 +15,14 @@ on:
-                    fi
-                done
-                cd ..
-            done
-
-            echo "Show snaps connections"
-            snap connections turtlebot3c-bringup
-            snap connections turtlebot3c-nav
-            snap connections turtlebot3c-teleop
-            sleep 2
-
-            echo "Show snaps services"
-            snap services turtlebot3c-bringup
-            sleep 2
-
-            echo "Set the simulation parameter"
-            sudo snap set turtlebot3c-bringup simulation=true
-            sudo snap set turtlebot3c-nav simulation=true
-            sleep 10
-
-            sudo snap start turtlebot3c-teleop.core
-            check_existence rosnode mux 30
-
-            ## set gui arg to false with sed since the arg is not exposed
-            sudo sed -i "s/\(<arg name=\"gui\" value=\"\).*\(\"\/>\)/\1false\2/" /opt/ros/noetic/share/turtlebot3_gazebo/launch/turtlebot3_empty_world.launch
-
-            ## Launch turtlebot simulation
-            TURTLEBOT3_MODEL=waffle_pi roslaunch turtlebot3_gazebo turtlebot3_empty_world.launch debug:=true &
-            pid=$!
-
-            echo "check simulation is running"
-            check_publishing scan 30
-            check_publishing joint_states 30
-
-            ## test core robot_state_publisher
-            check_existence rostopic tf 30
-            check_existence rosnode robot_state_publisher 30
-
-            ## test mapping node
-            sudo snap start turtlebot3c-nav.mapping
-            check_existence rostopic map 30
-            check_existence rosnode turtlebot3_slam_gmapping 30
-            echo "moving the robot"
-            timeout 10s rostopic pub -r 10 /cmd_vel geometry_msgs/Twist -- '[0.0, 0.0, 0.0]' '[1.0, 0.0, 0.0]'
-            sudo snap stop turtlebot3c-nav.mapping
-
-            ## check map has been created
-            timeout=30
-            start_time=$(date +%s)
-            while true; do
-              echo "check map exists.."
-              sudo bash -c 'cd /root/snap/turtlebot3c-nav/common/map/ && ls | grep -q ".yaml"'
-              if [ $? -eq 0 ]; then
-                echo "map found - starting navigation"
-                break
-              fi
-              current_time=$(date +%s)
-              elapsed_time=$((current_time - start_time))
-              if [ $elapsed_time -ge $timeout ]; then
-                echo "Map not found"
-                exit 1
-              fi
-              sleep 1
-            done
-
-            sudo snap start turtlebot3c-nav.navigation
-            check_existence rosnode move_base 30
-            check_existence rostopic nav_vel 30
-            sudo snap stop turtlebot3c-nav.navigation
-
-            kill $pid
-            sudo snap stop turtlebot3c-bringup
-
-            EOF
-
-            bash test-script.sh
+      snap-name: turtlebot3c-bringup
+      publish: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
```

Comment:
please ask for a new channel in the store and use it here as a parameter

Reply:
Added the track variable. Channels created https://forum.snapcraft.io/t/track-request-turtlebot3c-bringup-noetic-humble/41035/2